Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -13139,10 +13139,19 @@ void WebPageProxy::substitutionsPanelIsShowing(CompletionHandler<void(bool)>&& c
     completionHandler(TextChecker::substitutionsPanelIsShowing());
 }
 
-void WebPageProxy::showCorrectionPanel(AlternativeTextType panelType, const FloatRect& boundingBoxOfReplacedString, const String& replacedString, const String& replacementString, const Vector<String>& alternativeReplacementStrings)
+void WebPageProxy::showCorrectionPanel(AlternativeTextType panelType, const FloatRect& boundingBoxOfReplacedString, String&& replacedString, String&& replacementString, Vector<String>&& alternativeReplacementStrings, FrameIdentifier rootFrameID)
 {
-    if (RefPtr pageClient = this->pageClient())
-        pageClient->showCorrectionPanel(panelType, boundingBoxOfReplacedString, replacedString, replacementString, alternativeReplacementStrings);
+    RefPtr pageClient = this->pageClient();
+    if (!pageClient)
+        return;
+
+    convertRectToMainFrameCoordinates(boundingBoxOfReplacedString, rootFrameID, [pageClient = WTFMove(pageClient), panelType = WTFMove(panelType), replacedString = WTFMove(replacedString),
+        replacementString = WTFMove(replacementString), alternativeReplacementStrings = WTFMove(alternativeReplacementStrings)](std::optional<WebCore::FloatRect> convertedBoundingBox) {
+        if (!convertedBoundingBox)
+            return;
+
+        pageClient->showCorrectionPanel(panelType, convertedBoundingBox.value(), replacedString, replacementString, alternativeReplacementStrings);
```

Comment:
nit: I tend to like *convertedBoundingBox instead of convertedBoundingBox.value() because it makes it more clear that it is something that should've already been checked.  You do the check so it's fine either way.